File: backend/internal/constants/curing_state.go
package constants

type CuringState string

const (
	CuringPrepared         CuringState = "prepared"
	CuringPoured           CuringState = "poured"
	CuringActive           CuringState = "curing"
	CuringSuspended        CuringState = "suspended"
	CuringThresholdReached CuringState = "threshold_reached"
	CuringClosed           CuringState = "closed"
)

var curingTransitions = map[CuringState]map[CuringState]struct{}{
	CuringPrepared: {
		CuringPoured:    {},
		CuringSuspended: {},
	},
	CuringPoured: {
		CuringActive:    {},
		CuringSuspended: {},
	},
	CuringActive: {
		CuringThresholdReached: {},
		CuringSuspended:        {},
	},
	CuringThresholdReached: {
		CuringClosed:    {},
		CuringSuspended: {},
	},
	CuringSuspended: {
		CuringActive: {},
		CuringClosed: {},
	},
}

func (s CuringState) Valid() bool {
	_, ok := curingTransitions[s]
	return ok || s == CuringClosed
}

func CanTransitionCuring(from, to CuringState) bool {
	next, ok := curingTransitions[from]
	if !ok {
		return false
	}
	_, ok = next[to]
	return ok
}

func CuringStateValues() []string {
	return []string{
		string(CuringPrepared), string(CuringPoured), string(CuringActive),
		string(CuringSuspended), string(CuringThresholdReached), string(CuringClosed),
	}
}

File: frontend/src/types/enums/curing-state.ts
export const CURING_STATES = ['prepared', 'poured', 'curing', 'suspended', 'threshold_reached', 'closed'] as const
export type CuringState = (typeof CURING_STATES)[number]

export const CURING_STATE_LABELS: Record<CuringState, string> = {
  prepared: '待浇筑',
  poured: '已浇筑',
  curing: '养护中',
  suspended: '已暂停',
  threshold_reached: '达到阈值',
  closed: '已关闭',
}

export const CURING_TRANSITIONS: Record<CuringState, CuringState[]> = {
  prepared: ['poured', 'suspended'],
  poured: ['curing', 'suspended'],
  curing: ['threshold_reached', 'suspended'],
  threshold_reached: ['closed', 'suspended'],
  suspended: ['curing', 'closed'],
  closed: [],
}
